### ISSUE TITLE
Should aria-labelledby be prohibited when its reference is invalid or visible?

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4118]
I think `aria-prohibited-attr` is a little too strict for a rule that tests for WCAG. First up, just like how we don't fail `aria-prohibited-attr` when `aria-label` is empty, I think we should pass when `aria-labelledby` results in an empty string:

```html
<div aria-labelledby="d1"><!-- Pass this --></div>
<div id="d1" aria-labelledby="nowhere"><!-- Pass this --></div>
```

Closely related, I think that if the referenced element exists and is in the accessibility tree axe should report that as incomplete, similar to how it reports incomplete when the element has content. 

```html
<div aria-labelledby="h1"><!-- Incomplete this, h1 is visible --></div>
<h1 id="h1">Hello world</h1>
<div aria-labelledby="h2"><!-- Fail this, h2 is hidden --></div>
<h2 id="h2" aria-hidden="true">Hello world</h2>
```

## Optional

When we look at the subtree of the element to decide if its a fail or incomplete, we could check if the text of the label exists in the subtree, in which case instead of reporting it as incomplete we could quite reasonably pass this. That would cut down on the amount of incompletes this rule generates.

**Comment by straker** [comment]
This ticket is similar to https://github.com/dequelabs/axe-core/issues/5180 and can be done in the same batch (also after https://github.com/dequelabs/axe-core/pull/5259 is done). 

To summarize all the changes needed:

```html
<!-- pass because name is empty (already does this) -->
<div aria-label=""></div>

<!-- pass because name is empty -->
<div aria-labelledby="empty"></div>
<div id="empty"></div>

<!-- incomplete because name is not empty and named element is visible to screen readers -->
<div aria-labelledby="visible-content"></div>
<div id="visible-content">im visible</div>

<!-- fail because name is not empty and named element is not visible to screen readers -->
<div aria-labelledby="h2"></div>
<h2 id="h2" aria-hidden="true">Hello world</h2>
```
